### PR TITLE
Fix linter warnings

### PR DESCRIPTION
**Pull request by ericgaspar**
- *Fix linter warnings*

[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/kresus_ynh%20PR86%20(ericgaspar)/)

### CODE REVIEW

File: scripts/backup [hunk 1]
```diff
@@ -19,44 +19,48 @@ ynh_abort_if_errors
...
 #=================================================
+ynh_print_info --message="Loading installation settings..."
 
 app=$YNH_APP_INSTANCE_NAME
 
-final_path=$(ynh_app_setting_get "$app" final_path)
-domain=$(ynh_app_setting_get "$app" domain)
-
-db_name="$app"
+final_path=$(ynh_app_setting_get --app=$app --key=final_path)
```
Comment (nicofrand) [inline scripts/backup:26]: Where do you get that? The documentation does not look like this: https://yunohost.org/#/packaging_apps_arguments_management_en. Is the current way deprecated?

File: scripts/install [hunk 2]
```diff
@@ -42,85 +41,77 @@ fi
...
 test ! -e "$final_path" || ynh_die "This path already contains a folder"
 
-# Normalize the url path syntax
-path_url=$(ynh_normalize_url_path "$path_url")
-
-# Check web path availability
-ynh_webpath_available "$domain" "$path_url"
 # Register (book) web path
-ynh_webpath_register "$app" "$domain" "$path_url"
+ynh_webpath_register --app=$app --domain=$domain --path_url=$path_url
```
Comment (nicofrand) [inline scripts/install:45]: Is the path "normalization" + availability check done by `ynh_webpath_register` now?

File: scripts/install [hunk 3]
```diff
@@ -134,59 +125,61 @@ virtualenv --python=python3 --system-site-packages "${final_path}/venv"
...
 ynh_script_progression --message="Installing app"
+
 ynh_use_nodejs
 (
 	cd "$final_path"
 	chown -R "$app": "$final_path"
 
 	# ynh_npm only exists for YunoHost 3.8+
-	"$nodejs_path/npm" install --production --unsafe-perm
+	ynh_exec_warn_less "$nodejs_path/npm" install --production --unsafe-perm
```
Comment (nicofrand) [inline scripts/install:136]: If the manifests now require Yunohost 3.8.1 maybe I should use `ynh_npm` now?

File: scripts/install [hunk 4]
```diff
@@ -134,59 +125,61 @@ virtualenv --python=python3 --system-site-packages "${final_path}/venv"
...
 )
 
 #=================================================
 # Setup Kresus postgresql database
 #=================================================
+ynh_script_progression --message="Creating a PostgreSQL database..."
+
+db_name=$(ynh_sanitize_dbid --db_name=$app)
+db_user=$db_name
```
Comment (nicofrand) [inline scripts/install:145]: Are we sure there won't be any whitespace in it?

File: scripts/install [hunk 5]
```diff
@@ -134,59 +125,61 @@ virtualenv --python=python3 --system-site-packages "${final_path}/venv"
...
+db_pwd=$(ynh_app_setting_get --app=$app --key=psqlpwd)
```
Comment (nicofrand) [inline scripts/install:146]: How does that work now that it is set before the db setup? And why do we need to set it again line 149?

File: scripts/install [hunk 6]
```diff
@@ -134,59 +125,61 @@ virtualenv --python=python3 --system-site-packages "${final_path}/venv"
...
-ynh_replace_string "__DB_PASSWORD__"   "$db_pwd"             "$final_path/config.ini"
-
+ynh_replace_string --match_string="__PORT__"        --replace_string="$port"       --target_file="$final_path/config.ini"
+ynh_replace_string --match_string="__PATH__"        --replace_string="$path_url"   --target_file="$final_path/config.ini"
+ynh_replace_string --match_string="__FINALPATH__"   --replace_string="$final_path" --target_file="$final_path/config.ini"
+ynh_replace_string --match_string="__SALT__"        --replace_string="$salt"       --target_file="$final_path/config.ini"
+ynh_replace_string --match_string="__APP__"         --replace_string="$app"        --target_file="$final_path/config.ini"
+ynh_replace_string --match_string="__DOMAIN__"      --replace_string="$domain"     --target_file="$final_path/config.ini"
+ynh_replace_string --match_string="__DB_NAME__"     --replace_string="$db_user"    --target_file="$final_path/config.ini"
+ynh_replace_string --match_string="__DB_USER__"     --replace_string="$db_name"    --target_file="$final_path/config.ini"
```
Comment (nicofrand) [inline scripts/install:167]: Pre-existing but those two lines are inverted (`__DB_NAME__` should use `$db_user` and not `$db_name` and vice-versa)

File: scripts/install [hunk 7]
```diff
@@ -197,20 +190,19 @@ ynh_add_systemd_config
...
 
 #=================================================
 # RELOAD NGINX
 #=================================================
+ynh_script_progression --message="Reloading NGINX web server..."
 
-ynh_script_progression --message="Reloading nginx"
-service "$app" start
-service nginx reload
+ynh_systemd_action --service_name=nginx --action=reload
```
Comment (nicofrand) [inline scripts/install:205]: Don't we need to start the service like before (`service "$app" start`)?

File: scripts/restore [hunk 8]
```diff
@@ -59,74 +57,77 @@ ynh_restore_file "/etc/nginx/conf.d/$domain.d/$app.conf"
...
 #=================================================
 # RESTORE THE POSTGRESQL DATABASE
 #=================================================
-
 ynh_script_progression --message="Restoring database"
+
 ynh_psql_test_if_first_run
 ynh_print_OFF
-ynh_psql_setup_db "$db_user" "$db_name" "$db_pwd"
+ynh_psql_setup_db --db_user=$db_user --db_name=$db_name --db_pwd=$db_pwd
```
Comment (nicofrand) [inline scripts/restore:102]: Are we sure there won't be any whitespace in the variables?

File: scripts/install (hunk 7 above)
Comment (nicofrand) [reply scripts/install:205]: Like in the restoration script something like: `ynh_systemd_action --service_name=$app --action="start"`?

File: scripts/upgrade [hunk 9]
```diff
@@ -63,74 +67,63 @@ ynh_clean_setup () {
...
 
+ynh_exec_warn_less ynh_install_app_dependencies $pkg_dependencies
 
 # Now that postgresql is installed, check the db exists and the user is set up
 ynh_script_progression --message="Setting up database"
 if [ -z "$db_pwd" ]; then
 	ynh_psql_test_if_first_run
-	ynh_psql_setup_db "$db_user" "$db_name"
-	db_pwd=$(ynh_app_setting_get "$app" psqlpwd)
+	ynh_app_setting_set --app=$app --key=db_name --value=$db_name
```
Comment (nicofrand) [inline scripts/upgrade:83]: Why did you remove the `ynh_psql_setup_db`? Previous versions did not have a postgresql database and this is needed on ugprade.

**Review by nicofrand — COMMENTED** [review]
Hi,

Thank you a lot for the help! I have a few questions, basically to be able to better maintain the package later.

File: scripts/install (hunk 7 above)
Comment (ericgaspar) [reply scripts/install:213]: Yes indeed!

File: scripts/restore (hunk 8 above)
Comment (ericgaspar) [reply scripts/restore:102]: I am sticking to the **example_ynh** template. We normally don't get spaces in thos variables...
https://github.com/YunoHost/example_ynh/blob/25251b00d316c41ed1fef6df90312df60615b8da/scripts/restore#L107-L110

Also I don't know if `ynh_print_OFF` `ynh_print_ON` are useful here ...

File: scripts/install (hunk 7 above)
Comment (ericgaspar) [reply scripts/install:213]: yes! I forgot 😅

File: scripts/backup (hunk 1 above)
Comment (ericgaspar) [reply scripts/backup:26]: I follow the example_ynh template...

https://github.com/YunoHost/example_ynh/blob/25251b00d316c41ed1fef6df90312df60615b8da/scripts/backup#L31

File: conf/nginx.conf [hunk 10]
```diff
@@ -8,9 +9,8 @@ location __PATH__/ {
 }
 
 location ~ __PATH__/\.(css|js|png|jpe?g|svg|eot|woff2?)$ {
-	more_set_headers Cache-Control 'max-age=2592000, must-revalidate, public';
+	more_set_headers 'Cache-Control: max-age=2592000, must-revalidate, public';
```
Comment (nicofrand) [inline conf/nginx.conf:12]: Nice catch, I'll put it in another commit as this is not just lint but probably a fix!

**Review by nicofrand — CHANGES_REQUESTED** [review]

File: scripts/upgrade [hunk 11]
```diff
@@ -80,8 +80,8 @@ ynh_exec_warn_less ynh_install_app_dependencies $pkg_dependencies
 ynh_script_progression --message="Setting up database"
 if [ -z "$db_pwd" ]; then
 	ynh_psql_test_if_first_run
-	ynh_psql_setup_db --db_user=$db_user --db_name=$db_name
 	db_pwd=$(ynh_app_setting_get --app=$app --key=psqlpwd)
```
Comment (nicofrand) [inline scripts/upgrade:83]: This should be after `ynh_psql_setup_db`

File: scripts/install [hunk 12]
```diff
@@ -143,10 +143,11 @@ ynh_script_progression --message="Creating a PostgreSQL database..."
 
 db_name=$(ynh_sanitize_dbid --db_name=$app)
 db_user=$db_name
-db_pwd=$(ynh_app_setting_get --app=$app --key=psqlpwd)
+
 ynh_app_setting_set --app=$app --key=db_name --value=$db_name
 
 ynh_psql_test_if_first_run
+db_pwd=$(ynh_app_setting_get --app=$app --key=psqlpwd)
```
Comment (nicofrand) [inline scripts/install:150]: This should be after `ynh_psql_setup_db`, and `ynh_psql_setup_db` should not pass `--db_pwd=$db_pwd`

**Review by nicofrand — CHANGES_REQUESTED** [review]

File: scripts/backup [hunk 13]
```diff
@@ -19,44 +19,48 @@ ynh_abort_if_errors
...
-ynh_script_progression --message="Backing up systemd service..."
 ynh_backup "/etc/systemd/system/$app.service"
 
-ynh_script_progression --message="Backup of $app completed" --last
\ No newline at end of file
+#=================================================
+# END OF SCRIPT
+#=================================================
+
+ynh_print_info --message="Backup of $app completed"
```
[the remark this answers is not in the document]
Comment (ericgaspar) [reply scripts/backup:66]: `ynh_print_info` doesn't take `--last`getopt

File: scripts/upgrade [hunk 14]
```diff
@@ -63,74 +67,63 @@ ynh_clean_setup () {
...
 
+ynh_exec_warn_less ynh_install_app_dependencies $pkg_dependencies
 
 # Now that postgresql is installed, check the db exists and the user is set up
 ynh_script_progression --message="Setting up database"
 if [ -z "$db_pwd" ]; then
 	ynh_psql_test_if_first_run
-	ynh_psql_setup_db "$db_user" "$db_name"
-	db_pwd=$(ynh_app_setting_get "$app" psqlpwd)
+	ynh_psql_setup_db --db_user=$db_user --db_name=$db_name
```
Comment (nicofrand) [inline scripts/upgrade:83]: We should probably sanitize the db_name before (like on install)? `db_name=$(ynh_sanitize_dbid --db_name=$app)`

Comment (ericgaspar) [reply scripts/upgrade:83]: Why not set it like in example_ynh template ('if [ -z "$db_name" ]; then' ...)
https://github.com/YunoHost/example_ynh/blob/25251b00d316c41ed1fef6df90312df60615b8da/scripts/upgrade#L50-L54

Comment (nicofrand) [reply scripts/upgrade:83]: This will store the db_name in the app's setting but that won't create it I think…

Comment (nicofrand) [reply scripts/upgrade:83]: See https://github.com/YunoHost/example_ynh/blob/25251b00d316c41ed1fef6df90312df60615b8da/scripts/upgrade#L46.

Our case is that Kresus 0.15 did not have a postgresql database, so on ugprade we need to create it, not just save the db_name in setting.

**Comment by ericgaspar** [comment]
I might be missing specific things for this package, so feel free to modify the code as you like

**Comment by nicofrand** [comment]
Thanks a lot! I'll double check everything, separate fixes from real lint and will merge ASAP!

File: scripts/install [hunk 15]
```diff
@@ -42,85 +41,80 @@ fi
...
 
 #=================================================
 # NGINX CONFIGURATION
 #=================================================
+ynh_script_progression --message="Configuring NGINX web server..."
 
 # Create a dedicated nginx config
 ynh_add_nginx_config
 
-if [ "$path_url" != "/" ]
```
Comment (nicofrand) [inline scripts/install:97]: One last question: can you explain why this is not needed anymore please?

Comment (ericgaspar) [reply scripts/install:97]: That's included in NGINX helper.
https://github.com/YunoHost/yunohost/blob/1279755463100e6c4a0ba7d4b50f8f39658c8eac/data/helpers.d/nginx#L60

**Comment by nicofrand** [comment]
I split the commits in several (to separate fixes from lint), rebased and squashed in !88. Thanks again!
